src/keycard/keycard.cljs

```diff
@@ -147,17 +148,23 @@
 
 (defn generate-and-load-key
   [{:keys [mnemonic pin on-success on-failure]}]
-  (.. status-keycard
-      (generateAndLoadKey mnemonic pin)
-      (then on-success)
-      (catch on-failure)))
+  (native-module/validate-mnemonic
+   mnemonic
+   (fn []
```

Comment:
im not sure how it works, it seems like you don't check the result of validation?

Reply:
Sorry, I realized that later. I'll make the necessary changes.